tinymce-single-react-ui/src/external/wp-blocks/editable/index.js

```diff
@@ -0,0 +1,261 @@
+/**
+ * External dependencies
+ */
+import { createElement, Component } from 'wp-elements';
+import tinymce from 'tinymce';
+import { isEqual, last } from 'lodash';
+
+function initialize( node, inline, onSetup ) {
+	if ( ! node ) {
+		return;
+	}
+
+	const config = {
+		target: node.querySelector( '[contenteditable=true]' ),
+		theme: false,
+		inline: true,
+		toolbar: false,
+		skin_url: '//s1.wp.com/wp-includes/js/tinymce/skins/lightgray',
+		entity_encoding: 'raw',
+		setup: onSetup,
+		formats: {
+			strikethrough: { inline: 'del' }
+		}
+	};
+
+	if ( inline ) {
+		config.valid_elements = '#p,br,b,i,strong,em,del,a[href|target]';
+	}
+
+	tinymce.init( config );
+}
+
+export default class EditableComponent extends Component {
+	static defaultProps = {
+		onChange: () => {},
+		splitValue: () => {},
+		onFocusChange: () => {},
+		onType: () => {},
+		initialContent: '',
+		inline: false,
+		single: false,
+	};
+
+	componentDidMount() {
+		initialize( this.node, this.props.inline, this.onSetup );
+		if ( this.props.focusConfig ) {
+			this.focus();
+		}
+	}
+
+	updateContent() {
+		// This could not be called on each content change, it used to change the cursor position
+		let bookmark;
+		if ( this.props.focusConfig ) {
+			bookmark = this.editor.selection.getBookmark( 2, true );
+		}
+		this.editor.setContent( this.props.content );
+		if ( this.props.focusConfig ) {
+			this.editor.selection.moveToBookmark( bookmark );
+		}
+	}
```

Comment:
Regarding a 'controlled' component
- we could try it out (can restore the selection, for example)
- this would probably mean though that tiny plugins that change the dom will fail
- it seems less intrusive and risky to drive it the unmanaged way you did:
  -- using the Tiny API -  adding event handlers for state updates, and calling ExecCommand to get things done in the content